community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_2/Compatibility.scala

```diff
@@ -89,84 +89,97 @@ trait Compatibility[C <: CompilerContext] {
-        case CypherExecutionMode.profile => ProfileModev3_2
-        case CypherExecutionMode.normal => NormalModev3_2
-      }
-      exceptionHandler.runSafely {
-        val innerParams = typeConversions.asPrivateMap(params)
-        val innerResult = inner.run(queryContext(transactionalContext), innerExecutionMode, innerParams)
-
-        new ExecutionResult(
-          new ClosingExecutionResult(
-            transactionalContext.tc.executingQuery(),
-            new ExecutionResultWrapper(innerResult,
-                                       inner.plannerUsed,
-                                       inner.runtimeUsed,
-                                       preParsingNotifications,
-                                       Some(offSet)),
-            exceptionHandler.runSafely)
-        )
-      }
-    }
+  def info(message: String) {
+    log.info(message)
+  }
+}
+
+class ExecutionPlanWrapper(inner: ExecutionPlan_v3_2,
+                           transactionalContext: TransactionalContextWrapperV3_3,
+                           preParsingNotifications: Set[org.neo4j.graphdb.Notification],
+                           offSet: frontend.v3_2.InputPosition,
+                           searchMonitor: IndexSearchMonitor,
+                           executionMonitor: QueryExecutionMonitor)
+  extends ExecutionPlan {
 
-    def isPeriodicCommit: Boolean = inner.isPeriodicCommit
-
-    def isStale(lastCommittedTxId: LastCommittedTxIdProvider, ctx: TransactionalContextWrapperV3_3): Boolean =
-      inner.isStale(lastCommittedTxId, TransactionBoundGraphStatistics(ctx.readOperations))
-
-    override def plannerInfo: PlannerInfo = {
-      import scala.collection.JavaConverters._
-      new PlannerInfo(inner.plannerUsed.name, inner.runtimeUsed.name, inner.plannedIndexUsage.map {
-        case SchemaIndexSeekUsage(identifier, label, propertyKeys) =>
-          val labelId = transactionalContext.readOperations.labelGetForName(label)
-          schemaIndexUsage(identifier, labelId, label, propertyKeys: _*)
-        case SchemaIndexScanUsage(identifier, label, propertyKey) =>
-          val labelId = transactionalContext.readOperations.labelGetForName(label)
-          schemaIndexUsage(identifier, labelId, label, propertyKey)
-        case LegacyNodeIndexUsage(identifier, index) => explicitIndexUsage(identifier, "NODE", index)
-        case LegacyRelationshipIndexUsage(identifier, index) => explicitIndexUsage(identifier, "RELATIONSHIP", index)
-      }.asJava)
-    }
 
-    override def run(transactionalContext: TransactionalContextWrapperV3_3, executionMode: CypherExecutionMode, params: MapValue): Result = {
-      var map: mutable.Map[String, Any] = mutable.Map[String, Any]()
-      params.foreach(new BiConsumer[String, AnyValue] {
-        override def accept(t: String, u: AnyValue): Unit = map.put(t, valueHelper.fromValue(u))
-      })
+  private def queryContext(transactionalContext: TransactionalContextWrapperV3_3) = {
```

Comment:
Why do we need to change 3.2 so much more than 3.1 and 2.3?

Reply:
The reason is that ExecutionPlanWrapper was defined as in inner class of Compatibility (across all versions). This is not really needed and made it impossible for us to write the unit test. We decided to extract it to its own class for 3.2, in order to be able to test it. 

We could do the same for the other versions, but we could also do that on demand, if needed, which I think makes sense.